modules/ingestor/src/service/advisory/csaf/creator.rs

```diff
@@ -297,6 +323,34 @@ impl<'a> StatusCreator<'a> {
         Ok(())
     }
 
+    fn from_purl_version_scheme(purl: &Purl) -> VersionScheme {
+        match purl.ty.as_str() {
+            "maven" => VersionScheme::Maven,
+            "npm" => VersionScheme::Semver,
+            "python" => VersionScheme::Python,
+            "rpm" => VersionScheme::Rpm,
+            "semver" => VersionScheme::Semver,
+            _ => VersionScheme::Generic,
+        }
+    }
```

Comment:
It looks like this function is pretty similar to https://github.com/trustification/trustify/blob/106ee32185925cb367dd61a0046c85b8f8df0c73/modules/ingestor/src/service/advisory/cve/loader.rs#L308-L319
so it might be worth thinking about having some `impl From` in `VersionScheme`.

WDYT?